spring-data-microservice-transaction-sample/README.md

```diff
@@ -465,84 +457,76 @@ if (updatedCreditTotal > customer.creditLimit) {
 
 ```java
-execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Validate", () -> {
-  // Validate the transaction
-  customerRepository.validate();
-  return Empty.getDefaultInstance();
-});
+customerRepository.validateTransactionOnParticipant(request.getTransactionId());
 ```
 
-In the `commit` endpoint of the Customer Service, the microservice resumes and commits the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L140-L144)):
+In the `commit` endpoint of the Customer Service, it resumes and commits the transaction with `ScalarDbTwoPcRepository.commitTransactionOnParticipant()`. For reference, see [`CustomerService.java`](customer-service/src/main/java/sample/customer/CustomerService.java).
 
 ```java
-execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Commit", () -> {
-  // Commit the transaction
-  customerRepository.commit();
-  return Empty.getDefaultInstance();
-});
+customerRepository.commitTransactionOnParticipant(request.getTransactionId());
 ```
 
 ### Error handling
 
-If an error occurs during the transaction, the transaction will be automatically rolled back by using `executeTwoPcTransaction()`. The implementation to invoke the `rollback` gRPC endpoint of the Customer Service also needs to be passed as a parameter to the API with other ones (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+If an error happens while executing a transaction, the transaction will be automatically rolled back in both the local Order Service and the remote Customer Service by `ScalarDbTwoPcRepository.executeTwoPcTransaction()`. The implementation to invoke the `rollback` gRPC endpoint of the Customer Service also needs to be passed as a parameter to the API with other ones. For reference, see [`OrderService.java`](order-service/src/main/java/sample/order/OrderService.java).
 
 ```java
-Collections.singletonList(
-  RemotePrepareCommitPhaseOperations.createSerializable(
-    this::callPrepareEndpoint,
-    this::callValidateEndpoint,
-    this::callCommitEndpoint,
-    this::callRollbackEndpoint
+TwoPcResult<PlaceOrderResponse> result = orderRepository.executeTwoPcTransaction(txId ->{
+    ...
+  },
+
+  Collections.singletonList(
+    RemotePrepareCommitPhaseOperations.createSerializable(
+      this::callPrepareEndpoint,
+      this::callValidateEndpoint,
+      this::callCommitEndpoint,
+      this::callRollbackEndpoint
+    )
   )
-)
+);
 ```
 
-In the `rollback` endpoint of the Customer Service, the microservice resumes and rolls back the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L149-L153)):
+In the `rollback` endpoint of the Customer Service, it resumes and rolls back the transaction. For reference, see [`CustomerService.java`](customer-service/src/main/java/sample/customer/CustomerService.java).
```

Comment:
Does `it` mean ``the `rollback` endpoint``? If so, how about we change this sentence as follows?:

```suggestion
In the `rollback` endpoint of the Customer Service, the endpoint resumes and rolls back the transaction. For reference, see [`CustomerService.java`](customer-service/src/main/java/sample/customer/CustomerService.java).
```

Reply:
@josh-wong Thanks for the suggestions! I'm committing them.